### ISSUE TITLE
serviceaccount "jenkins-x-controllerbuild" not found

### ISSUE THREAD

**Issue by mbakker2000** [jenkins-x/jx#6079]
### Summary 

trying to clean install Jenkins X on minikube (2CPU, 4GB mem and 20GB disk)

`jx boot`

the builder pod seems to fail on not finding the service account

### Steps to reproduce the behavior

trying to clean install Jenkins X on minikube (2CPU, 4GB mem and 20GB disk)

`jx boot`

### Expected behavior

a working environment

### Actual behavior

I get this output in the console (Terminal)

```
unable to recognize "/var/folders/mn/xhdxxwq91mn4cc_z7psk8s5m0000gn/T/helm-template-workdir-003227207/jenkins-x/output/namespaces/jx/env/charts/prow/templates/part0-pipelinerunner-deployment.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "/var/folders/mn/xhdxxwq91mn4cc_z7psk8s5m0000gn/T/helm-template-workdir-003227207/jenkins-x/output/namespaces/jx/env/charts/prow/templates/part0-tide-deployment.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "/var/folders/mn/xhdxxwq91mn4cc_z7psk8s5m0000gn/T/helm-template-workdir-003227207/jenkins-x/output/namespaces/jx/env/charts/tekton/templates/part0-controller.yaml": no matches for kind "Deployment" in version "apps/v1beta1"'
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step helm apply --boot --remote --name jenkins-x --provider-values-dir ../kubeProviders' command in directory 'env', output: ''
```

When checking the minikube interface, `minikube dashboard`

```
Error creating: pods "jenkins-x-controllerbuild-7fbb6d754c-" is forbidden: error looking up service account jx/jenkins-x-controllerbuild: serviceaccount "jenkins-x-controllerbuild" not found
```

When checking the service account 
```
kind: Secret
apiVersion: v1
metadata:
  name: jenkins-x-controllerbuild-token-wmzg9
  namespace: jx
  selfLink: /api/v1/namespaces/jx/secrets/jenkins-x-controllerbuild-token-wmzg9
  uid: e0a1c32b-9d9f-4c0d-ad92-8bbf66c665ad
  resourceVersion: '4314'
  creationTimestamp: '2019-11-11T21:33:54Z'
  annotations:
    kubernetes.io/service-account.name: jenkins-x-controllerbuild
    kubernetes.io/service-account.uid: 40629ed6-9faf-48d8-96c6-565532c5f35f
data:
  ca.crt: >-
    Some data
  namespace: ang=
  token: >-
    Some data
type: kubernetes.io/service-account-token

```

### Jx version

The output of `jx version` is:

NAME               VERSION
jx                 2.0.974
Kubernetes cluster v1.16.2
kubectl            v1.16.2
helm client        Client: v2.14.3+g0e7f3b6
git                2.15.2 (Apple Git-101.1)
Operating System   Mac OS X 10.13.6 build 17G9016

using version stream URL: https://github.com/jenkins-x/jenkins-x-versions.git and git ref: v1.0.198

### Jenkins type

- [x ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

minikube cluster

### Operating system / Environment

MacOS High Sierra 10.13.6

**Comment by mbakker2000** [comment]
I upgraded everything got the same issue

`Upgrade OSX`

`brew upgrade`

`jx version`

```
NAME               VERSION
jx                 2.0.976
Kubernetes cluster v1.16.2
kubectl            v1.16.2
helm client        Client: v2.14.3+g0e7f3b6
git                2.21.0 (Apple Git-122.2)
Operating System   Mac OS X 10.15.1 build 19B88
```

**Comment by mbakker2000** [comment]
After more cleaning started with an empty home brew 

`brew remove --force $(brew list).`

and reverted this fix

https://github.com/jenkins-x/jx/issues/6018#issuecomment-550007510

Everything seems to work probably

**Comment by mbakker2000** [comment]
no still no fix same issue

**Comment by kozy** [comment]
Getting similar errors

```
Setting up prow config into namespace jx
Installing tekton into namespace jx
? A local Jenkins X versions repository already exists, pulling the latest: Yes
WARNING: 
retrying after error:failed to run 'kubectl apply --recursive -f /tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx -l jenkins.io/chart-release=tekton --namespace jx --validate=false' command in directory '', output: 'podsecuritypolicy.policy/tekton-pipelines created
clusterrole.rbac.authorization.k8s.io/tekton-pipelines created
clusterrolebinding.rbac.authorization.k8s.io/tekton-pipelines-jx created
secret/knative-git-user-pass created
serviceaccount/tekton-pipelines created
serviceaccount/tekton-bot created
customresourcedefinition.apiextensions.k8s.io/clustertasks.tekton.dev created
customresourcedefinition.apiextensions.k8s.io/pipelines.tekton.dev created
customresourcedefinition.apiextensions.k8s.io/pipelineruns.tekton.dev created
customresourcedefinition.apiextensions.k8s.io/pipelineresources.tekton.dev created
customresourcedefinition.apiextensions.k8s.io/tasks.tekton.dev created
customresourcedefinition.apiextensions.k8s.io/taskruns.tekton.dev created
service/tekton-pipelines-controller created
service/tekton-pipelines-webhook created
clusterrole.rbac.authorization.k8s.io/tekton-bot created
clusterrolebinding.rbac.authorization.k8s.io/tekton-bot-jx created
role.rbac.authorization.k8s.io/tekton-bot created
rolebinding.rbac.authorization.k8s.io/tekton-bot created
configmap/config-artifact-bucket created
configmap/config-artifact-pvc created
configmap/config-defaults created
configmap/config-entrypoint created
configmap/config-logging created
unable to recognize "/tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx/tekton/templates/part0-controller.yaml": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "/tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx/tekton/templates/part0-webhook.yaml": no matches for kind "Deployment" in version "apps/v1beta1"'
? A local Jenkins X versions repository already exists, pulling the latest: Yes
error: configuring and installing Prow: installing Prow: failed to install Tekton: after 2 attempts, last error: failed to run 'kubectl apply --recursive -f /tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx -l jenkins.io/chart-release=tekton --namespace jx --validate=false' command in directory '', output: 'podsecuritypolicy.policy/tekton-pipelines configured
clusterrole.rbac.authorization.k8s.io/tekton-pipelines unchanged
clusterrolebinding.rbac.authorization.k8s.io/tekton-pipelines-jx unchanged
secret/knative-git-user-pass unchanged
serviceaccount/tekton-pipelines unchanged
serviceaccount/tekton-bot configured
customresourcedefinition.apiextensions.k8s.io/clustertasks.tekton.dev unchanged
customresourcedefinition.apiextensions.k8s.io/pipelines.tekton.dev unchanged
customresourcedefinition.apiextensions.k8s.io/pipelineruns.tekton.dev unchanged
customresourcedefinition.apiextensions.k8s.io/pipelineresources.tekton.dev unchanged
customresourcedefinition.apiextensions.k8s.io/tasks.tekton.dev unchanged
customresourcedefinition.apiextensions.k8s.io/taskruns.tekton.dev unchanged
service/tekton-pipelines-controller unchanged
service/tekton-pipelines-webhook unchanged
clusterrole.rbac.authorization.k8s.io/tekton-bot unchanged
clusterrolebinding.rbac.authorization.k8s.io/tekton-bot-jx unchanged
role.rbac.authorization.k8s.io/tekton-bot unchanged
rolebinding.rbac.authorization.k8s.io/tekton-bot unchanged
configmap/config-artifact-bucket configured
configmap/config-artifact-pvc unchanged
configmap/config-defaults unchanged
configmap/config-entrypoint unchanged
configmap/config-logging unchanged
unable to recognize "/tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx/tekton/templates/part0-controller.yaml": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "/tmp/helm-template-workdir-270455700/tekton/output/namespaces/jx/tekton/templates/part0-webhook.yaml": no matches for kind "Deployment" in version "apps/v1beta1"'

```

**Comment by dpateltech** [comment]
I believe this has to do with the kubernetes apps/v1beta1 deprecation, more information here: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

**Comment by jiangytcn** [comment]
@kozy fyi,  I'm working on the fix for compatible with kube api 1.16+
